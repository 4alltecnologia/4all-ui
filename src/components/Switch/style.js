import styled from 'styled-components';
import { flexPosition, fontStyle, globalStyle } from '../../styles/mixins';
import theme from '../../styles/variables';

export const Switch = styled.div`
  ${globalStyle()}
  ${flexPosition({ align: 'center' })};
`;

export const SwitchBlock = styled.label`
  position: relative;
  display: inline-block;
  width: 41px;
  min-width: 41px;
  height: 24px;
  margin: 0;
`;

export const SwitchText = styled.span`
  ${fontStyle(
    theme.colors.GRAY_DARKER,
    theme.fontSizes.SIZE_7,
    theme.fontWeights.NORMAL,
  )}
  position: relative;
  margin-left: 10px;
  ${({ labelStyles }) => labelStyles};
`;

export const SwitchInput = styled.input`
  position: absolute;
  opacity: 0;
  :focus {
    ~ span {
      outline-width: 2px;
      outline-style: solid;
      outline-color: Highlight;
    }
  }
`;

export const Slider = styled.span`
  border-radius: 34px;
  cursor: ${({ disabled }) => (disabled ? 'default' : 'pointer')};
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  transition: transform 0.4s;
  background-color: ${({ checked }) => checked ? theme.colors.MAIN_COLOR : theme.colors.GRAY_LIGHT_2};
  ${({ customStyles }) => customStyles};

  :before {
    position: absolute;
    content: '';
    height: 16px;
    width: 16px;
    bottom: 4px;
    background-color: ${theme.colors.WHITE};
    transition: transform 0.4s;
    box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.12);
    border-radius: 50%;
    transform: ${({ checked }) => checked ? 'translateX(20px)' : 'translateX(4px)'};
    ${({ sliderStyles }) => sliderStyles};
  }
`;
