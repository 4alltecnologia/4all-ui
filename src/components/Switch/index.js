import React from 'react';
import PropTypes from 'prop-types';
import * as styles from './style';
import { GlobalStyle } from '../../styles/mixins';

const Switch = ({
  id,
  className,
  checked,
  disabled,
  onChange,
  customStyles,
  sliderStyles,
  labelStyles,
  children,
}) => (
  <styles.Switch>
    <GlobalStyle />
    <styles.SwitchBlock>
      <styles.SwitchInput
        id={id}
        className={`switch-4all ${className}`}
        checked={checked}
        type="checkbox"
        onChange={onChange}
        disabled={disabled}
      />
      <styles.Slider
        checked={checked}
        customStyles={customStyles}
        sliderStyles={sliderStyles}
        disabled={disabled}
      />
    </styles.SwitchBlock>
    <styles.SwitchText labelStyles={labelStyles}>{children}</styles.SwitchText>
  </styles.Switch>
);

Switch.propTypes = {
  id: PropTypes.string,
  className: PropTypes.string,
  checked: PropTypes.bool,
  disabled: PropTypes.bool,
  customStyles: PropTypes.object,
  sliderStyles: PropTypes.object,
  labelStyles: PropTypes.object,
  onChange: PropTypes.func,
  children: PropTypes.any,
};

Switch.defaultProps = {
  id: null,
  className: '',
  checked: false,
  disabled: false,
  onChange: null,
  customStyles: {},
  sliderStyles: {},
  labelStyles: {},
  children: '',
};

export default Switch;
