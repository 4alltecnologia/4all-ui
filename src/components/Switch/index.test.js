import React from 'react';
import { shallow } from 'enzyme';
import Switch from '.';

const onClick = jest.fn();

describe('<Switch />', () => {
  it('should render the component correctly', () => {
    const component = shallow(
      <Switch id="test" className="test" onChange={onClick} checked />,
    );

    expect(component).toMatchSnapshot();
  });
});
