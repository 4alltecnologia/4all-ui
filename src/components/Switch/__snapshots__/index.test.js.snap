// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Switch /> should render the component correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Switch
    checked={true}
    className="test"
    customStyles={null}
    disabled={false}
    id="test"
    labelStyles={null}
    onChange={[Function]}
    sliderStyles={null}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Unknown>
        <GlobalStyleComponent />
        <Unknown>
                <Unknown
                        checked={true}
                        className="switch-4all test"
                        disabled={false}
                        id="test"
                        onChange={[Function]}
                        type="checkbox"
                />
                <Unknown
                        checked={true}
                        customStyles={null}
                        disabled={false}
                        sliderStyles={null}
                />
        </Unknown>
        <Unknown
                labelStyles={null}
        />
</Unknown>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <GlobalStyleComponent />,
          <Unknown>
            <Unknown
                        checked={true}
                        className="switch-4all test"
                        disabled={false}
                        id="test"
                        onChange={[Function]}
                        type="checkbox"
            />
            <Unknown
                        checked={true}
                        customStyles={null}
                        disabled={false}
                        sliderStyles={null}
            />
</Unknown>,
          <Unknown
            labelStyles={null}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Unknown
                checked={true}
                className="switch-4all test"
                disabled={false}
                id="test"
                onChange={[Function]}
                type="checkbox"
/>,
              <Unknown
                checked={true}
                customStyles={null}
                disabled={false}
                sliderStyles={null}
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "className": "switch-4all test",
                "disabled": false,
                "id": "test",
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bxivhb",
                  "isStatic": true,
                  "rules": Array [
                    "
  position: absolute;
  opacity: 0;
  :focus {
    ~ span {
      outline-width: 2px;
      outline-style: solid;
      outline-color: Highlight;
    }
  }
",
                  ],
                },
                "displayName": "styled.input",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bxivhb",
                "target": "input",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "checked": true,
                "customStyles": null,
                "disabled": false,
                "sliderStyles": null,
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-ifAKCX",
                  "isStatic": false,
                  "rules": Array [
                    "
  position: absolute;
  cursor: ",
                    [Function],
                    ";
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: ",
                    "#E4E4E4",
                    ";
  -webkit-transition: transform .4s;
  transition: transform .4s;
  border-radius: 34px;
  background-color: ",
                    [Function],
                    ";
  ",
                    [Function],
                    ";
  
  :before{
    position: absolute;
    content: \\"\\";
    height: 16px;
    width: 16px;
    bottom: 4px;
    background-color: ",
                    "#FFFFFF",
                    ";
    -webkit-transition: transform .4s;
    transition: transform .4s;
    box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.12);
    border-radius: 50%;
    background-color: ",
                    [Function],
                    ";
    transform: ",
                    [Function],
                    ";
    -webkit-transform: ",
                    [Function],
                    ";
    -ms-transform: ",
                    [Function],
                    ";
    ",
                    [Function],
                    ";
  }
",
                  ],
                },
                "displayName": "styled.span",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-ifAKCX",
                "target": "span",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  position: relative;
  display: inline-block;
  width: 41px;
  min-width: 41px;
  height: 24px;
  margin: 0;
",
              ],
            },
            "displayName": "styled.label",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "label",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
            "labelStyles": null,
          },
          "ref": null,
          "rendered": "",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htpNat",
              "isStatic": false,
              "rules": Array [
                "
  position: relative;
  margin-left: 10px;
  ",
                [Function],
                ";
",
              ],
            },
            "displayName": "styled.span",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-htpNat",
            "target": "span",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  ",
            "
  display: flex;
  align-items: center;
  
  
",
            ";
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Unknown>
          <GlobalStyleComponent />
          <Unknown>
                    <Unknown
                              checked={true}
                              className="switch-4all test"
                              disabled={false}
                              id="test"
                              onChange={[Function]}
                              type="checkbox"
                    />
                    <Unknown
                              checked={true}
                              customStyles={null}
                              disabled={false}
                              sliderStyles={null}
                    />
          </Unknown>
          <Unknown
                    labelStyles={null}
          />
</Unknown>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <GlobalStyleComponent />,
            <Unknown>
              <Unknown
                            checked={true}
                            className="switch-4all test"
                            disabled={false}
                            id="test"
                            onChange={[Function]}
                            type="checkbox"
              />
              <Unknown
                            checked={true}
                            customStyles={null}
                            disabled={false}
                            sliderStyles={null}
              />
</Unknown>,
            <Unknown
              labelStyles={null}
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <Unknown
                  checked={true}
                  className="switch-4all test"
                  disabled={false}
                  id="test"
                  onChange={[Function]}
                  type="checkbox"
/>,
                <Unknown
                  checked={true}
                  customStyles={null}
                  disabled={false}
                  sliderStyles={null}
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "className": "switch-4all test",
                  "disabled": false,
                  "id": "test",
                  "onChange": [Function],
                  "type": "checkbox",
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-bxivhb",
                    "isStatic": true,
                    "rules": Array [
                      "
  position: absolute;
  opacity: 0;
  :focus {
    ~ span {
      outline-width: 2px;
      outline-style: solid;
      outline-color: Highlight;
    }
  }
",
                    ],
                  },
                  "displayName": "styled.input",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-bxivhb",
                  "target": "input",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "checked": true,
                  "customStyles": null,
                  "disabled": false,
                  "sliderStyles": null,
                },
                "ref": null,
                "rendered": null,
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "sc-ifAKCX",
                    "isStatic": false,
                    "rules": Array [
                      "
  position: absolute;
  cursor: ",
                      [Function],
                      ";
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: ",
                      "#E4E4E4",
                      ";
  -webkit-transition: transform .4s;
  transition: transform .4s;
  border-radius: 34px;
  background-color: ",
                      [Function],
                      ";
  ",
                      [Function],
                      ";
  
  :before{
    position: absolute;
    content: \\"\\";
    height: 16px;
    width: 16px;
    bottom: 4px;
    background-color: ",
                      "#FFFFFF",
                      ";
    -webkit-transition: transform .4s;
    transition: transform .4s;
    box-shadow: 0 1px 1px 0 rgba(0, 0, 0, 0.24), 0 0 1px 0 rgba(0, 0, 0, 0.12);
    border-radius: 50%;
    background-color: ",
                      [Function],
                      ";
    transform: ",
                      [Function],
                      ";
    -webkit-transform: ",
                      [Function],
                      ";
    -ms-transform: ",
                      [Function],
                      ";
    ",
                      [Function],
                      ";
  }
",
                    ],
                  },
                  "displayName": "styled.span",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "sc-ifAKCX",
                  "target": "span",
                  "toString": [Function],
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                },
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bwzfXH",
                "isStatic": true,
                "rules": Array [
                  "
  position: relative;
  display: inline-block;
  width: 41px;
  min-width: 41px;
  height: 24px;
  margin: 0;
",
                ],
              },
              "displayName": "styled.label",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-bwzfXH",
              "target": "label",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "",
              "labelStyles": null,
            },
            "ref": null,
            "rendered": "",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htpNat",
                "isStatic": false,
                "rules": Array [
                  "
  position: relative;
  margin-left: 10px;
  ",
                  [Function],
                  ";
",
                ],
              },
              "displayName": "styled.span",
              "foldedComponentIds": Array [],
              "render": [Function],
              "styledComponentId": "sc-htpNat",
              "target": "span",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  ",
              "
  display: flex;
  align-items: center;
  
  
",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
